### DISCUSSION TITLE
Ticket collection/verification from barcode/QR code in a mobile app

### DISCUSSION THREAD

**Discussion by mikcatta** [notJust-dev/notjustdev#252] [Ideas]
Hello All from the UK !

Someone on our local FB group asked about any apps on the stores for collecting tickets (at the entrance to an event). I presume this would involve scanning a QR code/barcode using the phone camera. I believe there is a bit more to it, such as a backend to prevent duplicates and also section for tasks such as taking payment ahead of time & producing reports. 

As part of this, years ago, I wrote an app in C# for scanning the barcodes from books, looking them up online, selecting a category (fiction, non-fiction, children's) and then using a bit of web scrapping to upload the details to a 3rd party website, where it would allocate it a unique reference, which would then be grabbed from the HTML doc, parsed and printed on label printer, all in real-time. All with their permission of course :). 

Sounds like their might be elements that could be combined from the two ideas and put into a stream for our wonderful Vadim.

**Comment by VadimNotJustDev** [comment]
Hey @mikcatta 

Thanks for the idea. Looks like a great project to dive into.

The interesting part is generating QR codes for each ticket, and then scanning them with the app to validate.
